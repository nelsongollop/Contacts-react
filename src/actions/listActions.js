import axios from "axios";

export function getContacts () {
    return{
        type: "GET_CONTACTS",
        payload: new Promise((resolve, reject) => {
            axios.get('http://localhost:3001/api/v1/contacts')
                .then(
                    (result) => {
                        resolve(result.data)
                    },
                    (error) => {
                        reject(error)
                    }
                )
        })
    }
}

export function filter(keywords) {
    return {
        type: "FILTER_LIST",
        payload: keywords
    }
}

export function addContact(contact){
    return{
        type: "ADD_CONTACT",
        payload: contact
    }
}

