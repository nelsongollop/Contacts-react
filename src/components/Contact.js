import React, {useEffect} from 'react'
import { Field, reduxForm } from 'redux-form'
import TextField from "@material-ui/core/TextField";
import Avatar from "@material-ui/core/Avatar";
import Button from '@material-ui/core/Button'
import {connect} from 'react-redux'

const renderTextField = ({input, meta, label}) => (
    <TextField
        {...input}
        id="standard-name"
        label={label}
        margin="normal"
        fullWidth={true}
        helperText={meta.touched ? meta.error : ""}
        FormHelperTextProps={{error: true}}
        onKeyPress={e => {
            if (e.key === 'Enter') e.preventDefault();
        }}
    />
)

let ContactForm = props => {
    const { handleSubmit } = props
    useEffect(() => {
        props.initialize(props.state.selected)
        props.change('id', props.state.selected.id)
    }, [props.state.selectedIndex])

    return (
        <div className="container">
            <div className="row justify-content-md-center">
                <Avatar/>
            </div>
            <div className="row justify-content-md-center">
                <div className="col-md-8">
                    <form onSubmit={handleSubmit}>
                        <Field name="first_name" component={renderTextField} label="First Name"/>
                        <Field name="last_name" component={renderTextField} label="Last Name"/>
                        <Field name="phone" component={renderTextField} label="Phone"/>
                        <Field name="email" component={renderTextField} label="Email"/>

                        <Button className="float-right mt-5" variant="contained" color="secondary" type="submit">
                            {props.state.update ? "Update" : "Create"}
                        </Button>
                    </form>
                </div>
            </div>
        </div>
    )
}

ContactForm = reduxForm({
    form: 'contact'
})(ContactForm)

const mapStateToProps = (state) => {
    return {
        state : state.list
    }
}

export default connect(mapStateToProps)(ContactForm)