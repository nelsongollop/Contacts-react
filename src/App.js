import React, {useState} from "react";
import List from './components/List'
import NavBar from './components/NavBar'
import Contact from "./components/Contact";
import Unselected from "./components/dumb/unselected";
import {connect} from "react-redux";
import {getContacts, setSelected, setUpdate} from "./actions/listActions";
import {Snackbar} from "@material-ui/core";
import {Alert} from "@material-ui/lab";
import * as API from './api/requests'

const App = (props) => {
    const [snackOpen, setOpen] = useState(false)

    //submit callback function passed as prop to the form
    const onSubmit = values => {
        console.log(values)
        if (props.state.update){
            API.updateContact(values)
                .then(
                    (result) => {
                        props.getContacts()
                        setOpen(true)
                        props.setUpdate(null)
                        props.setSelected(null)
                    },
                    (error) => console.log(error)
                )
        } else {
            API.createContact(values)
                .then(
                    (result) => {
                        props.getContacts()
                        setOpen(true)
                        props.setUpdate(null)
                        props.setSelected(null)
                    },
                    (error) => console.log(error)
                )
        }
    }

    const handleClose = (event, reason) => {
        if (reason === 'clickaway') {
            return;
        }

        setOpen(false);
    };

    return(
        <div className="container-fluid h-100 d-flex flex-column">
            <div className="row">
                <NavBar/>
            </div>
            <div className="row h-100">
                <div className="col-md-6 no-padding d-flex">
                    <List/>
                </div>
                <div className="col-md-6 no-padding vertical-center">
                    {props.state.update != null ?
                        <Contact onSubmit={onSubmit} update={props.state.update}/> :
                        <Unselected/>
                    }
                </div>
            </div>
            <Snackbar open={snackOpen}
                      autoHideDuration={6000}
                      onClose={handleClose}
                      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
            >
                <Alert onClose={handleClose} color="success">
                    Your contact was successfully added
                </Alert>
            </Snackbar>
        </div>
    )
}

const mapStateToProps = (state) => {
    return{
        state: state.list
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getContacts: () => {
            dispatch(getContacts())
        },
        setSelected: (index) => {
            dispatch(setSelected(index))
        },
        setUpdate: (update) => {
            dispatch(setUpdate(update))
        },
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
