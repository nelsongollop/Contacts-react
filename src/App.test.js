import React from 'react';
import {mockComponent} from "react-dom/test-utils";
import { render } from '@testing-library/react';
import App from './App';

import renderer from 'react-test-renderer';
import {Provider} from 'react-redux'
import Contact from "./components/Contact";
import configureStore from 'redux-mock-store';
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const mockStore = configureStore([]);

test('Test enviroment is working properly', () => {
  expect(true).toBeTruthy();
});

test('Contact renders', () => {
  expect(<Contact/>).toBeTruthy()
})

describe('My Connected React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      list: {
        selectedIndex: null,
        selected: {},
        updating: false
      }
    });
    component = renderer.create(
        <Provider store={store}>
          <Contact />
        </Provider>
    );
  });
  it('should render a new contact form', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('The button should say create', () => {
    const instance = component.root;
    const button = instance.findByType("button");
    expect(button.props.children).toHaveValue("Create");
  });
  // it('should dispatch an action on button click', () => {
  // });
});

import Navbar from './components/NavBar'
import {setUpdate, setSelected, filter} from "./actions/listActions";

describe('Navbar', () => {
  let store;
  let wrapper;
  beforeEach(() => {
    store = mockStore();
    store.dispatch = jest.fn();

    wrapper = shallow(
        <Provider store={store}>
          <Navbar />
        </Provider>
    );
  });

  it('should render a Navbar', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('The button should change the state', () => {
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });

    expect(store.dispatch).toHaveBeenCalledTimes(2);
    expect(store.dispatch).toHaveBeenCalledWith(
        setSelected(null)
    );
    expect(store.dispatch).toHaveBeenCalledWith(
        setUpdate(false)
    );
  });

  it('The textfield should change the state', () => {
    renderer.act(() => {
      component.root.findByType('button').props.onChange();
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
        filter()
    );
  });
});
