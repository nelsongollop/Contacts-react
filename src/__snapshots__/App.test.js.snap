// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render a new contact form 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-8"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            First Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="first_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Last Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="last_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Phone
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="phone"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained float-right mt-5 MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Create
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-8"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            First Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="first_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Last Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="last_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Phone
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="phone"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained float-right mt-5 MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Create
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`My Connected React-Redux Component should render with given state from Redux store 2`] = `
<div
  className="container"
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-8"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            First Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="first_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Last Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="last_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Phone
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="phone"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained float-right mt-5 MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Create
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Navbar should render a Navbar 1`] = `
<div
  className="makeStyles-root-118"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Contacts
      </h6>
      <div
        className="makeStyles-search-120"
      >
        <div
          className="makeStyles-searchIcon-121"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="MuiInputBase-root makeStyles-inputRoot-122"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            className="MuiInputBase-input makeStyles-inputInput-123"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search…"
            type="text"
          />
        </div>
      </div>
      <div
        className="makeStyles-title-119"
      />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          New Contact
        </span>
      </button>
    </div>
  </header>
</div>
`;

exports[`Navbar should render a new contact form 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <div
    className="row justify-content-md-center"
  >
    <div
      className="col-md-8"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            First Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="first_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Last Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="last_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Phone
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="phone"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onDragStart={[Function]}
          onDrop={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-name"
            id="standard-name-label"
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-name"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
          
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained float-right mt-5 MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Create
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
